Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -2232,8 +2296,28 @@ def visit_ModuleNode(self, node):
 
     def visit_FuncDefNode(self, node):
         node.local_scope.infer_types()
-        node.body = node.body.analyse_expressions(node.local_scope)
-        self.visitchildren(node)
+        for n in range(2):
+            # potentially have two goes - replacing a fastcall tuple star_arg
+            # if too many coercions are generated. Check after each step
```

Comment:
Maybe. The problem with a _single_ tuple object is that there can be more than one fastcall tuple in a function since slices create new fastcall tuple objects, and I think that's useful.

It might be better to try what I've done with fastcall dict, where it can have two states (a coerced and non-coerced state). It'd make the objects themselves a little less lightweight to use, but it'd avoid the coercion counting.

I'll look if this could work.